crates/common/download/src/download.rs

```diff
@@ -159,17 +159,16 @@ impl Downloader {
 
 fn create_file_and_try_pre_allocate_space(
     file_path: &Path,
-    file_len: usize,
+    file_len: u64,
```

Comment:
We can not use the u64 here because thin-edge also has to be supported on 32bit arm processors as well.

Reply:
That's exactly the reason why this patch was made: If `reqwest::Response::content_length()` returns a number bigger than `u32`, the code would truncate it to `usize`, which would be 32 bit.

Reply:
So, how it will compile on 32bit systems?.

Reply:
I believe the problem previously was not exactly 32-bit arm but rather something in musl preventing the correct build.
@matthiasbeyer would you mind making sure check other targets including `musl`, I think one of the workflows has the list of currently supported builds.

Reply:
Yes, the issue was on musl.

Reply:
So for `cargo c --target aarch64-unknown-linux-gnu` and `cargo c --target aarch64-unknown-linux-musl` cargo succeeded for me.
The others failed because `ring` does not build on this target with my gcc... I guess there's setup on my side is missing.

Reply:
You should try 32 bit musl
cross build  --target armv7-unknown-linux-musleabihf

Reply:
Took me a while because I had to build gcc and musl for testing this out, but yes, this `cargo check --target armv7-unknown-linux-musleabihf`s as expected on my machine.

Reply:
I ran test as below

cross test --target armv7-unknown-linux-musleabihf

Find below issue. 

error[E0308]: mismatched types
   --> crates/common/download/src/download.rs:246:58
    |
246 |         let usable_disk_space = tmpstats.blocks_free() * tmpstats.block_size();
    |                                                          ^^^^^^^^^^^^^^^^^^^^^ expected `u64`, found `u32`

error[E0277]: cannot multiply `u64` by `u32`
   --> crates/common/download/src/download.rs:246:56
    |
246 |         let usable_disk_space = tmpstats.blocks_free() * tmpstats.block_size();
    |                                                        ^ no implementation for `u64 * u32`
    |
    = help: the trait `Mul<u32>` is not implemented for `u64`

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `download` due to 2 previous errors
warning: build failed, waiting for other jobs to finish..

Reply:
Hm, I wonder why this was not catched in my `cargo check --tests` run, because this is not armv7 or musl specific.